src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/IVSTypeScriptFormattingServiceImplementation.cs

```diff
@@ -14,7 +15,7 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
         int TabSize,
         int IndentSize);
 
-    internal interface IVSTypeScriptFormattingServiceImplementation
+    internal interface IVSTypeScriptFormattingServiceImplementation : ILanguageService
```

Comment:
📝 The only change required from the TypeScript side is the implementation of this interface will need to change:

```patch
-[Export(typeof(IVSTypeScriptFormattingServiceImplementation)), Shared]
+[ExportLanguageService(typeof(IVSTypeScriptFormattingServiceImplementation), InternalLanguageNames.TypeScript), Shared]

Reply:
Why is this necessary? It seems you'd only need this for the CreateLanguageService call, when that code could just do an [Import(AllowDefault = true)] and just do the null check that way.

Reply:
> when that code could just do an [Import(AllowDefault = true)] and just do the null check that way.

This is another viable option. The approach of implementing `ILanguageService` is a better match for our overall design, but if TypeScript is already using/exporting the interface in the current form I can switch it for compatibility.

Reply:
@zkat to confirm that this interface is not already implemented

Reply:
I don't see us using it anywhere

Reply:
It's being implemented in https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS/pullrequest/373465; rather than reset that, let's just do the AllowDefault pattern which it's already expecting.